JabRef/jabref
# Change optional fields tabs

Fixes #4688

The changes we have made are the followings : 
-  delete OtherFieldsTab.java class, OptionalFields2Tab.java files from the jabref
   why-> as we were merging the optionalfields tab and otherFieldsTab, these both java class files existence were not required        anymore. Hence got deleted. 

- Deleted +136 to +151 lines of code from FieldEditorTab.java 
   why->these lines of the codes were responsible the column division. As a result of this merged text field were looking ugly(see the fig-1.1 and fig-1.2)
![fig 11](https://user-images.githubusercontent.com/88136301/170812695-6f9bb01f-27a7-4b4a-ad7c-3daf3ca190a7.png)
![fig 1 1 column issue](https://user-images.githubusercontent.com/88136301/170812698-6d49b2a8-63e7-4288-b38c-4d17be5e6318.png)



-  updated the code of the OptionalsTabBase.java 
   why-> we wrote the codes for merging the field of the requested tab. 

---> Tittle pr : Fix issue #4688 by UofA


- [x] Screenshots added in PR description (for UI changes)


How to see the fix: 
1. Create a new lib 
![image](https://user-images.githubusercontent.com/88136301/170811030-8144a6a8-ec16-4d7c-9a30-10205f6f8a30.png)

2. Click on the "new Entry" as marked on the snapshot and then add the doi of any article or your preferred the document type.
![image](https://user-images.githubusercontent.com/88136301/170811228-c943d026-7f44-48c5-8f67-fb2140d93116.png)

3. Last thing is just double click on the entry to see the changes. 
![image](https://user-images.githubusercontent.com/88136301/170811273-f9eb0396-e17f-416f-9e1b-887ec8dc8098.png)
